client/my-sites/earn/home.tsx

```diff
@@ -262,6 +263,51 @@ const Home: FunctionComponent< ConnectedProps > = ( {
 		};
 	};
 
+	/**
+	 * Return the content to display in the Paid Newsletter card based on the current plan.
+	 *
+	 * @returns {object} Object with props to render a PromoCard.
+	 */
+	const getPaidNewsletterCard = () => {
+		const cta = isFreePlan
+			? {
+					text: translate( 'Unlock this feature' ),
+					action: () => {
+						trackUpgrade( 'any-paid-plan', 'premium-content' );
+						page( `/plans/${ selectedSiteSlug }` );
+					},
+			  }
+			: {
+					text: translate( 'Learn how to get started' ),
+					action: () => {
+						trackCtaButton( 'learn-paid-newsletters' );
+						page( `https://wordpress.com/support/paid-newsletters/` );
```

Comment:
There is however going to be a race condition between the tracks event firing and the navigation event if the two are seperated like this.

Reply:
We also shouldn't be using page() with external links. It'll throw a history API error:

<img width="1712" alt="Screen Shot 2020-06-05 at 2 35 49 PM" src="https://user-images.githubusercontent.com/1270189/83924621-e5e22900-a739-11ea-8a89-54a62a8bd7d5.png">

Reply:
Let's use an External link component `import ExternalLink from 'components/external-link';` with an href and an onClick set to record the tracks event.

Reply:
If this ends up needing more refactor work to allow this, another option is checking if window is available in the current env and using a window.location call instead.

Reply:
To a look at this; since these card methods return prop objects instead of actual components, using `ExternalLink` would require either converting everything to return components, or adding a flag or something to signal the render method to conditionally use `ExternalLink` – all kinda messy.

> another option is checking if window is available in the current env and using a window.location call instead

@donlair This would not require the refactoring, and would look something like this:

```js
action: () => {
	trackCtaButton( 'learn-paid-newsletters' );
	if ( window && window.location ) {
		window.location.href = 'https://wordpress.com/support/paid-newsletters/';
	}
},
```

Reply:
yup a window call is fine here if it's too messy to refactor

Reply:
Thanks @kwight! I think my latest commit implements your suggestion properly. Could you take a look?